### DISCUSSION TITLE
Edit Add Node menu

### DISCUSSION THREAD

**Discussion by Durman** [nortikin/sverchok#4721] [Ideas]
We can add extra item to the Add Node menu which would allowed to include/exclude categories from the menu. When edit mode of the menu is on it shows all categories with check boxes which define categories to show when edit mode is off.

![image](https://user-images.githubusercontent.com/28003269/197941888-c9028f94-f816-40bb-a2e8-a2feb36d318b.png)

**Comment by portnov** [comment]
But if you tick one of the items, the menu will disappear, so to tick another item you will need to open the menu again?
In such case IMHO it would be better to move this to settings, to be visible as panel or dialog, not as pop-up menu.

**Reply by Durman** [reply]
Agree, it's not convenient in this way. I guess it can be a pop-up dialog instead. Something similar to Switcher UI.

![image](https://user-images.githubusercontent.com/28003269/197958355-4f23f652-d5df-4c70-8b21-f15dd039aeea.png)

**Comment by portnov** [comment]
Another concern is that I doubt many people will want to enable/disable one particular category.
IMHO usually you want to switch between "toolboxes" / "workspaces" (in Freecad terminology), for example:
* "Mesh" toolbox shows the following node categories:
  * Generator
  * Transforms
  * Analyzers
  * Modifiers
  * CAD
  * Number
  * Vector
* "Manifolds" (or better name) toolbox:
  * Curves
  * Surfaces
  * Fields
  * Solids
  * Number
  * Vector
* etc

Most probably we want content of these toolboxes to be configurable by user, especially considering that there are additional categories in addons that extend sverchok.
And we need fast enough way to switch between these toolboxes.
